### PR TITLE
Enable access to valid JWT token which is not subscribed to API

### PR DESCRIPTION
**Pull request by nalakas**
## Purpose
Enable access to valid JWT token which is not subscribed to API

## Goals
 JWT token which is valid and not subscribed to any API can access to API in the micro-gateway.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal [hunk 1]
```diff
@@ -82,6 +83,25 @@ public type SubscriptionFilter object {
                                     return true;
                                 }
                             }
+                            }
+                            else
+                            {
```
Comment (chamilaadhi) [inline components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal]: fix formatting issue

File: components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal [hunk 2]
```diff
@@ -82,6 +83,25 @@ public type SubscriptionFilter object {
...
+                                authenticationContext.applicationId = "";
+                                authenticationContext.applicationName = "";
+                                authenticationContext.applicationTier = "Unlimited";
+                                authenticationContext.subscriber = "";
+                                authenticationContext.consumerKey = decodedPayload.consumerKey.toString();
+                                authenticationContext.apiTier = decodedPayload.apiTier.toString();
+                                authenticationContext.apiPublisher = "";
+                                authenticationContext.subscriberTenantDomain = "";
+                                runtime:getInvocationContext().attributes[KEY_TYPE_ATTR] = authenticationContext.keyType
+                                ;
```
Comment (chamilaadhi) [inline components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal]: fix formatting issue

File: components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal [hunk 3]
```diff
@@ -82,6 +83,25 @@ public type SubscriptionFilter object {
...
                             }
+                            }
+                            else
+                            {
+                                authenticationContext.authenticated = true;
+                                authenticationContext.tier = "Unlimited";
+                                authenticationContext.apiKey = jwtToken;
+                                authenticationContext.applicationId = "";
+                                authenticationContext.applicationName = "";
+                                authenticationContext.applicationTier = "Unlimited";
```
Comment (chamilaadhi) [inline components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal]: Use a constant for Unlimited

File: components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal (hunk 1 above)
Comment (nalakas) [reply components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal]: this code has revamped and formatted with 4d34de4

File: components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal (hunk 2 above)
Comment (nalakas) [reply components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal]: this code has revamped and formatted with 4d34de4

File: components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal (hunk 3 above)
Comment (nalakas) [reply components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal]: this code has revamped and formatted with 4d34de4

**Pull request merged**